tests/validation/test_validate_json.py

```
@@ -1,9 +1,30 @@
+def test_validation_error_hierarchy() -> None:
+    """Test that validation errors inherit from the validation base error."""
+    assert issubclass(ValidationInvalidSchemaNameError, ValidationError)
+    assert issubclass(ValidationInvalidJsonDataError, ValidationError)
+
+
+def test_validation_error_module_and_pickle() -> None:
+    """Test that validation errors have the public Python module path and can be pickled."""
+    err = ValidationInvalidJsonDataError("boom")
+
+    assert ValidationInvalidJsonDataError.__module__ == "pyavd_utils.validation"
+    assert type(pickle.loads(pickle.dumps(err))) is ValidationInvalidJsonDataError
```

Ruff (0.15.18)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)